idiorm.php

```diff
@@ -536,7 +536,15 @@ protected static function _log_query($query, $parameters, $connection_name, $que
                 $bound_query = $query;
             } else {
                 // Escape the parameters
-                $parameters = array_map(array(self::get_db($connection_name), 'quote'), $parameters);
+                foreach ($parameters as &$parameter) {
+                    if (is_string($parameters) === true) {
+                        $parameter = [self::get_db($connection_name), "quote"]($parameter);
```

Comment:
Probably it has to be:
`if (is_string($parameter) === true) {`

also dynamical method calling, may lead to non-existent method calling exception, 
and can be changed to 
`$parameter = self::get_db($connection_name)->quote($parameter);`

Reply:
yeah, u right but its still have no sense cz author dont care about project)) (in my separated version var has right typing)

Reply:
Hey @devjet 

could you say what error is it fixing? it is not clear

Reply:
> Hey @devjet
> 
> could you say what error is it fixing? it is not clear

check original comment for PR ⬆️ or read this [issue](https://github.com/j4mie/idiorm/issues/374)

Reply:
Ahhh I see, it's a deprecated message only. Ok then it will be a problem on php 9 only, maybe you could inform it on your title,  just to be clear it is fixing an deprecation message.

Reply:
> Ahhh I see, it's a deprecated message only. Ok then it will be a problem on php 9 only, maybe you could inform it on your title, just to be clear it is fixing an deprecation message.

in full case it not only disabling deprecated message , this changes make more cleanest SQL code - null as NULL, boolean as BOOLEAN( e.g. `TRUE` or `FALSE` instead of `'1'` and `''`)

Reply:
> cz author dont care about project

Just to point out, that it isn't true, he is replying when he is mentioned.

> in full case it not only disabling deprecated message , this changes make more cleanest SQL code - null as NULL, boolean as BOOLEAN( e.g. TRUE or FALSE instead of '1' and '')

well if you want to "clean" the code and introduce a different behaviour you definitely need to write test for it, and   it would be nice if you could explain the benefits of this.


> Output Deprecated: PDO::quote(): Passing null to parameter https://github.com/j4mie/idiorm/issues/1 ($string) of type string is deprecated in C:\VHOST\VHOST8\idiorm\vendor\j4mie\idiorm\idiorm.php on line 539

checking your message ... it is complain about null value for quote function ... in theory there is null values on `$parameter`, But the question is : should `$parameter` contain null values?

Reply:
heaven sake...
at the first: isuue created >2 year, PR >1 y. yeah, author dont care about proj 
second: code upgrd for disable message about deprecated passing (read original issue)
thrid: if u dont what this code do - read the code, no explanation for it

P.S.
if u wanna to some rhetoric - go to some chat. i just pushed updates for issue solution, no more. this repo look like abandoned, so this discussion have no sense

Reply:
@eerison idiorm has a problem with inserting null values
this is problem because one have to edit something in vendors directory to fix it.